### PR TITLE
[annotations] Add support for callouts to point text, rect text and picture items

### PR DESCRIPTION
**Pull request by nyalldawson**
Allows creating of callout lines (or bubble callouts) which link the text to a map location. To create a callout, users select the text item and then drag out the central x node to the desired callout end point.

Callout styles can be modified through the layer styling panel.

https://github.com/user-attachments/assets/1a5006c6-69cb-4179-a844-4b4b4dc876c8

### CODE REVIEW

File: src/gui/qgsrubberband.h [hunk 1]
```diff
@@ -187,7 +187,7 @@ class GUI_EXPORT QgsRubberBand : public QgsMapCanvasItem
      * Sets the width of the line. Stroke width for polygon.
      *  \param width The width for any lines painted for this rubberband
      */
-    void setWidth( int width );
+    void setWidth( double width );
```
Comment (nirvn) [inline src/gui/qgsrubberband.h:190]: Good move to go from int to double on these functions here.

File: src/core/annotations/qgsannotationitem.h [hunk 2]
```diff
@@ -272,6 +274,66 @@ class CORE_EXPORT QgsAnnotationItem
...
+     */
+    void setCallout( QgsCallout *callout SIP_TRANSFER );
+
+    /**
+     * Returns the callout's anchor geometry.
+     *
+     * The anchor dictates the geometry which the option item callout() should connect to. Depending on the
+     * callout subclass and anchor geometry type, the actual shape of the rendered callout may vary.
+     *
+     * The callout anchor geometry is in the parent layer's coordinate reference system.
```
Comment (nirvn) [inline src/core/annotations/qgsannotationitem.h:313]: For project annotations, that would be the project CRS? Might be worth expanding here.

Comment (nyalldawson) [reply src/core/annotations/qgsannotationitem.h:313]: It's always the layer's CRS. The main annotation layer for a project still has it's own CRS, which may or may not be the same as the project's CRS.

**Review by nirvn — APPROVED** [review]

File: src/core/annotations/qgsannotationitem.h (hunk 2 above)
Comment (nirvn) [reply src/core/annotations/qgsannotationitem.h:313]: @nyalldawson , fair enough.

**Pull request merged**